WordPress/src/main/java/org/wordpress/android/ui/plans/PlansActivity.java

```diff
@@ -135,18 +141,49 @@ public void onPause() {
         EventBus.getDefault().unregister(this);
     }
 
-    private void updatePurchaseUI(int position) {
-        Plan plan = getPageAdapter().getPlan(position);
-        boolean showPurchaseButton;
-        if (plan.isCurrentPlan()) {
-            showPurchaseButton = false;
-        } else {
-            // don't show the purchase button unless the plan at this position is "greater" than
-            // the current plan for this site
-            long currentPlanProductId = WordPress.wpDB.getPlanIdForLocalTableBlogId(mLocalBlogID);
-            showPurchaseButton = plan.isAvailable() && plan.getProductID() > currentPlanProductId;
+    /**
+     * The 'Buy' button should be available if the current plan is free, and the selected plan upgrade is available.
+     * @param position
+     * @return boolean - true if the current plan could be added to the blog.
+     */
+    private boolean isBuyButtonAvailable(int position) {
+        long currentPlanProductId = WordPress.wpDB.getPlanIdForLocalTableBlogId(mLocalBlogID);
+        if (!PlansUtils.isFreePlan(currentPlanProductId)) {
+            return false;
+        }
+        final Plan plan = getPageAdapter().getPlan(position);
+        return plan.isAvailable() && !plan.isCurrentPlan();
+    }
+
+    /**
+     * The 'Upgrade' button should be available if the current plan is NOT free,
+     * and the selected plan upgrade is available, and the current plan ID < new plan ID.
+     *
+     * Note: Not used now, but will be when we'll implement upgrade of plan within the app.
+     *
+     * @param position
+     * @return boolean - true if the current plan could be upgraded to the blog.
+     */
+    private boolean isUpgradeButtonAvailable(int position) {
+        long currentPlanProductId = WordPress.wpDB.getPlanIdForLocalTableBlogId(mLocalBlogID);
+        if (isBuyButtonAvailable(position) || PlansUtils.isFreePlan(currentPlanProductId)) {
+            return false;
         }
 
+        final Plan plan = getPageAdapter().getPlan(position);
+
+        // No downgrade!
+        if (PlansUtils.isFreePlan(plan) || currentPlanProductId > plan.getProductID()) {
```

Comment:
Rather than hard-code `currentPlanProductId > plan.getProductID()`, I think we should have a helper routine that checks if one plan is "greater" than another. We had this earlier on, and it was added to avoid sprinkling our code with assumptions that a greater plan ID meant a more expensive plan.